src/commando/core.py

```diff
@@ -1,2 +1,124 @@
+
+
+def _process_repo(
+    repo, command, commit_message, branch, pr_title, pr_body
+):  # pylint:disable=too-many-arguments
+
+    # Collect the results of all commands run for this repo.
+    results = []
+
+    pprint = partial(_pprint, repo)
+
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        run = partial(_run, tmpdirname, results)
+
+        try:
+            pprint("Cloning")
+            run(
+                [
+                    "gh",
+                    "repo",
+                    "clone",
+                    repo,
+                    tmpdirname,
+                    "--",
+                    # Ensure that the origin remote is named 'origin', the
+                    # default name could be overridden by the git config.
+                    "--origin",
+                    "origin",
+                    # Try to speed it up by not cloning tags since we don't need them.
+                    # Note that I don't think we can speed this up even more by
+                    # using a shallow clone: gh-pr-upsert needs the commits and
+                    # history in order to compare branches.
+                    "--no-tags",
+                ]
+            )
+
+            pprint("Creating branch")
+            run(
+                [
+                    "git",
+                    "switch",
+                    # Don't automatically set upstream configuration if there's
+                    # a remote tracking branch with the same name.
+                    "--no-guess",
+                    # Don't set upstream tracking configuratin.
+                    "--no-track",
+                    # Create a new branch or reset an existing one to origin/main.
+                    "--force-create",
+                    branch,
+                    "origin/main",
+                ]
+            )
+
+            pprint("Running command")
+            pprint(command)
+            run(command, shell=True)
+
+            pprint("Committing any changes")
+            run(["git", "add", "--all"])
+
+            run(
+                [
+                    "git",
+                    "commit",
+                    # Don't crash if there are no changes to commit.
+                    # If an empty commit is created gh-pr-upsert will do the
+                    # right thing and not create an empty PR.
+                    "--allow-empty",
+                    "--message",
+                    commit_message,
+                ]
+            )
+
+            pprint("Sending PR")
+            try:
+                result = run(["gh-pr-upsert", "--title", pr_title, "--body", pr_body])
+            except CalledProcessError as err:  # pragma: no cover
+                if err.returncode > 1:
+                    # Exit codes >1 from gh-pr-upsert are not a problem.
+                    pprint(err.stdout.rstrip())
+                else:
+                    raise
+            else:
+                pprint(result.stdout.rstrip())
```

Comment:
Commando doesn't generally print the outputs of all the commands it runs (`git clone`, `git switch`, even the given `--command`) because it'd be too noisy. `gh-pr-upsert` is the one command whose output we do print since it has minimal output that's useful to always see in the logs